lib/class-wp-theme-json-resolver.php

```diff
@@ -0,0 +1,383 @@
+<?php
+/**
+ * Process the different data sources for site-level
+ * config and offers and API to work with them.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Class that abstracts the processing
+ * of the different data sources.
+ */
+class WP_Theme_JSON_Resolver {
+
+	/**
+	 * Container for data coming from core.
+	 *
+	 * @var WP_Theme_JSON
+	 */
+	private static $core = null;
+
+	/**
+	 * Container for data coming from the theme.
+	 *
+	 * @var WP_Theme_JSON
+	 */
+	private $theme = null;
```

Comment:
$core and $user are static fields, while $theme is not static. Similarly get_core_origin and get_user_origin and static functions while get_theme_origin. What is the reason to have the difference between core and user and the theme?

Reply:
The idea here is that core & user won't change during a page load while theme can ― hence we can share core&user among all instances while the theme data is per instance. The reason why the theme data can change is that it uses existing theme support data, which can be filtered by 3rd parties ― hence it needs to be recalculated upon every call. Does this make sense?